src/components/motorcycleForm/addMotorcycle.js

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { createMotorcycle } from '../../redux/motorcycles/motorcycles';
+
+const AddMotorcycle = () => {
+  const getData = () => ({
+    model: '',
+    price: 0,
+    duration_months: 0,
+    description: '',
+    image: '',
+  });
+  const [formData, setFormData] = useState(getData);
+
+  const dispatch = useDispatch();
+
+  const currentUser = useSelector((state) => state.auth);
+
+  useEffect(() => {
+    localStorage.setItem('formData', JSON.stringify(formData));
+  }, [formData]);
+
+  const modelHandle = (e) => {
+    setFormData({ ...formData, model: e.target.value });
+  };
+
+  const priceHandle = (e) => {
+    setFormData({ ...formData, price: e.target.value });
+  };
+
+  const durationHandle = (e) => {
+    setFormData({ ...formData, duration_months: e.target.value });
+  };
+
+  const descriptionHandle = (e) => {
+    setFormData({ ...formData, description: e.target.value });
+  };
+
+  const imageHandle = (e) => {
+    setFormData({ ...formData, image: e.target.value });
+  };
+
+  const submitHandle = () => {
+    dispatch(createMotorcycle(formData, currentUser.currentUser.id));
+  };
+
+  return (
+    <div className="form">
+      <h2>Add your Motor</h2>
+
+      <form onSubmit={submitHandle} className="form">
```

Comment:
The submitHandle is a function and it should be called when an event is onSubmit event is triggered. I suggest you refactor it to behave as such.
You can find an example below.

`onSubmit={ (e) => submitHandle(e)}`